parser/ParseBlockProperties.hpp

```diff
@@ -0,0 +1,218 @@
+ *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *   See the License for the specific language governing permissions and
+ *   limitations under the License.
+ **/
+
+#ifndef QUICKSTEP_PARSER_PARSE_BLOCK_PROPERTIES_HPP_
+#define QUICKSTEP_PARSER_PARSE_BLOCK_PROPERTIES_HPP_
+
+#include <memory>
+#include <string>
+#include <vector>
+
+#include "parser/ParseString.hpp"
+#include "parser/ParseTreeNode.hpp"
+#include "utility/Macros.hpp"
+#include "utility/PtrList.hpp"
+
+namespace quickstep {
+
+/** \addtogroup Parser
+ *  @{
+ */
+
+/**
+ * @brief Parsed BlockPropertyItem which simply a key (property) and the corresponding value.
+ * @details The value may actually be a list of Parse strings. This is to cover the case where a 
+ *    user specifies a subset of columns to compress rather than using the keyword ALL 
+ */
+class ParseBlockPropertyItem : public ParseTreeNode {
+ public:
+
+  // string representation of the ALL SQL keyword
+  static const std::string AllString;
+
+  enum Property {
+    kCompress,
+    kSort,
+    kType
+  };
+
+  /**
+   * @brief Constructor.
+   *
+   * @param line_number Line number of the first token of this node in the SQL statement.
+   * @param column_number Column number of the first token of this node in the SQL statement.
+   * @param property describes which type of block property this is. Corresponds to a field of the 
+   *          StorageBlockLayoutDescription message
+   * @param value a parse string representing the assignment of the property
+   **/
+  ParseBlockPropertyItem( int line_number, 
+                          int column_number,
+                          Property property,
+                          ParseString* value)
+                                  : ParseTreeNode(line_number, column_number),
+                                    property_(property) {
+    values_.reset(new PtrList<ParseString>);
+    values_->push_back(value);
+  }
+
+  /**
+   * @brief Constructor.
+   *
+   * @param line_number Line number of the first token of this node in the SQL statement.
+   * @param column_number Column number of the first token of this node in the SQL statement.
+   * @param property describes which type of block property this is. Corresponds to a field of the 
+   *          StorageBlockLayoutDescription message
+   * @param values list of parse strings representing the assignment of the property
+   **/
+  ParseBlockPropertyItem( int line_number, 
+                          int column_number,
+                          Property property,
+                          PtrList<ParseString>* values)
+                                  : ParseTreeNode(line_number, column_number),
+                                    property_(property),
+                                    values_(values) { }
+
+  std::string getName() const override {
+    return "BlockPropertyItem";
+  }
+
+  Property property() const {
```

Comment:
Please doxygen all public methods. For very simple accessors like this, it can be something as simple as:
```
@return The type of property set by this item.
```